velox/functions/prestosql/registration/ComparisonFunctionsRegistration.cpp

```diff
@@ -66,6 +66,35 @@ void registerComparisonFunctions() {
       UnscaledLongDecimal,
       UnscaledLongDecimal,
       UnscaledLongDecimal>({"between"});
+  registerFunction<
```

Comment:
We are not handling the rescaling here thus the result is probably wrong when the scale does not match.